### ISSUE TITLE
[QUESTION]: Hyperparameter Optimization Takes Too Long (still running)

### ISSUE THREAD

**Issue by spyda90** [chemprop/chemprop#553]
Hi,

I am running hyperparameter optimization using **--num_fold 10** and -**-epochs 60**.
The calculation has been running for 100h and it is still going.
To give you an idea the same settings but with **--num_fold 1 (default)** and **--epochs 30** (default) last about 3h.

Reading the manual, I am wondering if the reason of the endless calulcation is that I have used **--split_type random** instead of *--split_type cv*.

What do you think?

Thank you in advance.

Francesco

**Comment by kevingreenman** [comment]
Have you checked the log files to see if they're still writing output (`tail -f verbose.log`)? This will give you an idea of the progress status or if something has gone wrong.

I don't see why `--split_type random` would last longer than `--split_type cv`, but if you're doing `--num_folds`>1, it doesn't make sense to use  `--split_type random` because you'd just be training an identical model 10 times for `--num_folds 10`.

**Comment by spyda90** [comment]
Hi Kevin,

Thank you for your answer!

I ran something like:
chemprop_hyperopt --data_path $Database --dataset_type regression --num_iters 100 --config_save_path ./hyperopt/hyperopt_100iter_epochs60.json --metric rmse --extra_metrics r2 --seed 0 --split_type cv --num_fold 10 --epochs 60

Usually I have just a JSON file as output, I don't know if and where there is a tmp folder with verbose.log (different from training, where I usually see it in the output folder).
But I am sure it is still running since I can see the verbose calculation in the terminal that is continue to generate models.

Francesco

Francesco

**Comment by kevingreenman** [comment]
The only thing I can think of is that using cross validation and the higher number of epochs is resulting in different performance of the models, such that the larger HPO run is tending toward larger models (higher depth, FFN size, hidden size, and FFN number of layers) than the smaller HPO run once it starts proposing new hyperparameters non-randomly. The larger models could be taking longer to train.

**Comment by spyda90** [comment]
Calculation has finished at the end with 4 days and 18h of CPU time.

Depth in hyperopt was higher with this settings (5 instead of 2). 

Thank you for your help!
